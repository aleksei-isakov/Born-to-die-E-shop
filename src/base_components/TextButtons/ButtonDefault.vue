<template>
  <BaseTextButton class="button-default" @click="onClickEmitEvent">
    <slot></slot>
  </BaseTextButton>
</template>

<script>
import { BaseTextButton } from '../';

export default {
  name: 'ButtonDefault',

  components: {
    BaseTextButton
  },

  methods: {
    onClickEmitEvent() {
      this.$emit('click');
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/variables.scss';
.button-default {
  padding: 6px 16px;
  background-color: unset;
  color: $primary;
  border: 1px solid $primary;
  &:hover {
    color: $primary-hover;
    border-color: $primary-hover;
  }
  &:disabled {
    border-color: $button-disabled-bg;
  }
}
</style>
