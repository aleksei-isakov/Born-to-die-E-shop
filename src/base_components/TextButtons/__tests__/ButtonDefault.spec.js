import { shallowMount, createLocalVue } from '@vue/test-utils';
import ButtonDefault from '../ButtonDefault.vue';

describe('ButtonDefault.vue', () => {
  let wrapper;

  beforeEach(() => {
    const localVue = createLocalVue();

    wrapper = shallowMount(ButtonDefault, {
      localVue
    });

    wrapper.vm.onClickEmitEvent();
    expect(wrapper.find('button')).toBeTruthy();
    expect(wrapper.is(ButtonDefault)).toBeTruthy();
  });

  it('renders a valid snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
