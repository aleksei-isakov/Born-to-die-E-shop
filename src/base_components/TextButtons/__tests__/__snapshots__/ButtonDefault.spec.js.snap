// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonDefault.vue renders a valid snapshot 1`] = `
<basetextbutton-stub
  class="button-default"
/>
`;
